controllers/ibu_controller.go

```diff
@@ -200,10 +200,28 @@ func (r *ImageBasedUpgradeReconciler) handleAbortOrFinalize(ctx context.Context,
 		switch idleCondition.Reason {
 		case string(utils.ConditionReasons.Aborting):
 			nextReconcile, err = r.handleAbort(ctx, ibu)
+			if err != nil {
```

Comment:
Status update is better done inside the stage handler functions. Also, returning error will make it discard any CR update and redo the same reconcile. Return nil if we want to give up on aborting.

Reply:
Do you think we should retry the aborting?

Reply:
and if we want to retry, when we should give up and go to failed state?

Reply:
Exactly, it's hard to say when to retry and for how many times or how long. For now, we can just give up on first error and improve error handling later, when we know better what kind of error is likely transient. The plan was to requeue with an interval so handleAbortFailure or handleFinalizeFailure will be triggered. Those two will check for some kind of "manual cleanup done" signal so that the IBU can still go back to idle.